### ISSUE TITLE
Wasteland Biome Causing Server Slowdown

### ISSUE THREAD

**Issue by Killasaurus-Warbear** [Deltric/Generations#94]
**What is the bug?**

Whenever I'm deep in a wasteland biome the server experiences severe slowdown after only a short period of time.  This causes frequent "Can't keep up! Did the system time change, or is the server overloaded? Running xxxxxms behind, skipping xxx tick(s)" warnings and has caused the server to stop responding for 60+ seconds and shut itself down once. If I /warp or /spawn while experiencing the lag the server immediately becomes responsive after I'm teleported. It also happens regardless of whether I'm I'm on the surface or underground in a wasteland biome. However, if I retreat to the edge of the biome and am within just a few chunks of another biome, the lag seems to resolve after a few seconds. As soon as I move back a few chunks into the wasteland the lag returns.

I'm running the server on a dedicated server in my home and connecting over my local network. I'm the only one connecting to the server.

**What are the steps to reproduce the bug?**

Entering a wasteland biome and waiting a minute or two. Lag starts fairly shortly after entering.

**What version of Pixelmon and Forge are you on?**

Pixelmon Generatiosn version 8.4.2
Forge version 14.23.5.2847

**Please provide any screenshots or crash reports if needed.**

**Comment by Dysth** [comment]
Hi, is your server running Sponge?
A quick fix would be to search "performance" in your servers pixelmon.hocon and add 1 to each of the 3 options

**Comment by Killasaurus-Warbear** [comment]
Yes, I'm running Sponge. I've increased each of the three "tickRate##Radius" settings in the performance sections, and will test later today to see if it improves the issue. I did try /pokekill last night to see if it was just mobs causing the issue and it didn't seem to help even when used a few times in a row.

**Comment by Dysth** [comment]
/config/sponge/global.conf
The biggest cause of lag would likely be Sponge's default world save config if you haven't changed this. By default, auto-save-interval and auto-player-save-interval are set to about 45 seconds. We recommend about 5 minutes. Time is in ticks, 20 ticks per second
If lag still persists, please try the attached file and instructions included
[lagtest.zip](https://github.com/Deltric/Generations/files/6320444/lagtest.zip)

**Comment by Killasaurus-Warbear** [comment]
None of those settings seemed to have any effect on the lag. Here's the link to the spark results:

https://spark.lucko.me/P0UsQlTRat


I didn't mention this in the original post because /pokekill wasn't helping, but the utilization in the spark viewer seems to indicate it may be related to the spawning process: There seems to be a significantly higher number of pokemon spawning in Wasteland biomes than other biomes and at a much faster pace, including duplicate totems spawning immediately next to each other. I've included a screenshot to demonstrate:

![2021-04-15_16 24 33_2](https://user-images.githubusercontent.com/50350483/114940387-64eac800-9e07-11eb-9e59-19ee259493f4.png)

**Comment by Dysth** [comment]
Yikes that's insane- I've never seen that before. Do you have any plugins that control spawn rates such as a custom legendary spawn system or anything similar? Although seems to be mainly related to generating Trainers. What's the rarity rates of your NPC Trainers?
If you wouldn't mind sharing your pixelmon.hocon file that should help me see as well. I'll do some testing shortly to see if I can replicate this Wasteland phenomenon

**Comment by Killasaurus-Warbear** [comment]
I don't recall modifying anything to do with spawn rates or trainers and the only plugin I'm running related to pixelmon at all is the Pokestops mod from the "sidemonds" part of the generations homepage. 

I've included my pixelmon.hocon

[pixelmon.hocon.txt](https://github.com/Deltric/Generations/files/6321324/pixelmon.hocon.txt)

**Comment by Dysth** [comment]
We've got a theory but could you provide your debug.log? Just make sure the effected area is loaded with you standing in it when you grab the log file

**Comment by Killasaurus-Warbear** [comment]
Sure, here's the debug.log

[debug.log](https://github.com/Deltric/Generations/files/6321386/debug.log)
